### PR TITLE
[Merged by Bors] - feat: Maximum principle for convex functions on an interval

### PR DESCRIPTION
**Pull request by YaelDillies**
Corollaries of existing lemmas.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Convex/Jensen.lean [hunk 1]
```diff
@@ -301,11 +301,35 @@ theorem ConvexOn.exists_ge_of_mem_convexHull (hf : ConvexOn 𝕜 (convexHull 
...
-then the eventual minimum of `f` on `convexHull 𝕜 s` lies in `s`. -/
+/-- **Minimum principle** for concave functions. If a function `f` is concave on the convex hull of
+`s`, then the eventual minimum of `f` on `convexHull 𝕜 s` lies in `s`. -/
 theorem ConcaveOn.exists_le_of_mem_convexHull (hf : ConcaveOn 𝕜 (convexHull 𝕜 s) f) {x}
     (hx : x ∈ convexHull 𝕜 s) : ∃ y ∈ s, f y ≤ f x :=
   ConvexOn.exists_ge_of_mem_convexHull (β := βᵒᵈ) hf hx
 #align concave_on.exists_le_of_mem_convex_hull ConcaveOn.exists_le_of_mem_convexHull
 
+/-- **Maximum principle** for convex functions on a segment. If a function `f` is convex on the
+segment `[x, y]`, then the eventual maximum of `f` on `[x, y]` is at `x` or `y`. -/
```
Comment (eric-wieser) [inline Mathlib/Analysis/Convex/Jensen.lean:312]: What's the meaning of `eventual` here?

Comment (YaelDillies) [reply Mathlib/Analysis/Convex/Jensen.lean:312]: https://en.wiktionary.org/wiki/eventual:
```
3. Possible, potential.
```

Comment (eric-wieser) [reply Mathlib/Analysis/Convex/Jensen.lean:312]: I guess my confusion stems from the fact this lemma doesn't really make any claim about the maximum; it only provides an upper bound.

Comment (YaelDillies) [reply Mathlib/Analysis/Convex/Jensen.lean:312]: It should be read as "If `f` reaches a maximum, then it must reach it at `a` or `b`".

Comment (eric-wieser) [reply Mathlib/Analysis/Convex/Jensen.lean:312]: I don't think that's a reasonable informalization of the lean statement though.

Comment (eric-wieser) [reply Mathlib/Analysis/Convex/Jensen.lean:312]: I guess my complaint is equally valid against the existing lemmas docstrings, and so this is our of scope for this PR

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**